src/cuda/cudnn.jl

```diff
@@ -1,7 +1,20 @@
 import CUDA.CUDNN: batchnorm, ∇batchnorm
 
-(BN::Flux.BatchNorm)(x::Union{CuArray{T,2},CuArray{T,4},CuArray{T,5}}, cache = nothing) where T<:Union{Float32, Float64} =
-  BN.λ.(batchnorm(BN.γ, BN.β, x, BN.μ, BN.σ², BN.momentum; cache = cache, alpha = 1, beta = 0, eps = BN.ϵ, training = Flux.istraining()))
+function (BN::Flux.BatchNorm)(x::Union{CuArray{T,2},CuArray{T,4},CuArray{T,5}},
+                              cache=nothing) where T<:Union{Float32, Float64}
+  
+  @assert BN.affine "BatchNorm: only affine=true supported on gpu"
+  @assert BN.track_stats "BatchNorm: only track_stats=true supported on gpu"
+  @assert length(BN.β) == size(x, ndims(x)-1) "BatchNorm: input has wronng number of channels"
```

Comment:
I think these are checks of input, not sanity checks on the program, so ideally would not use `@assert`
```suggestion
  BN.affine || throw(ArgumentError("BatchNorm: only affine=true supported on gpu"))
  BN.track_stats || throw(ArgumentError("BatchNorm: only track_stats=true supported on gpu"))
  length(BN.β) == size(x, ndims(x)-1) || throw(ArgumentError("BatchNorm: input has wronng number of channels"))
```